### ISSUE TITLE
Intermittent Yarn issues

### ISSUE THREAD

**Issue by rhlsthrm** [ConnextProject/indra#5]
We are seeing a variety of yarn related issues mostly from running `yarn start`. Some of the errors we have seen include:

```
warning machinomy > @types/commander@2.12.2: This is a stub types definition for commander (https://github.com/tj/commander.js). commander provides its own type definitions, so you don't need @types/commander installed!
warning machinomy > @machinomy/contracts > @machinomy/hdwallet-provider > web3-provider-engine > ethereumjs-block > merkle-patricia-tree > level-ws > xtend > object-keys@0.4.0:
[2/4] Fetching packages...
[3/4] Linking dependencies...
error An unexpected error occurred: "ENOENT: no such file or directory, lstat '/root/.cache/yarn/v4/npm-@machinomy-hdwallet-provider-1.3.1-a7ecf4b70dfb1f59d550856b8e1884ac0ff6591f/node_modules/@machinomy/hdwallet-provider/
```

```
=============
[Makefile] => Start building hub-node-modules
docker run --name=connext_buidler --tty --rm --user 501:20 --volume=/Users/rahul/Desktop/connext/connext/indra/modules/hub:/root connext_builder:dev 501:20 "yarn install --network-timeout 1000000"
Running command as user 501:20
yarn install v1.12.3
info No lockfile found.
[1/4] Resolving packages...
warning connext > web3 > web3-bzz > swarm-js > fs-promise@2.0.3: Use mz or fs-extra^3.0 with Promise Support
warning connext > web3 > web3-bzz > swarm-js > tar.gz@1.0.7: ⚠️  WARNING ⚠️ tar.gz module has been deprecated and your application is vulnerable. Please use tar module instead: https://npmjs.com/tar
warning machinomy > @types/commander@2.12.2: This is a stub types definition for commander (https://github.com/tj/commander.js). commander provides its own type definitions, so you don't need @types/commander installed!
error Received malformed response from registry for "mocha". The registry may be down.
info Visit https://yarnpkg.com/en/docs/cli/install for documentation about this command.
Same user, skipping permission fix
make: *** [hub-node-modules] Error 1
error Command failed with exit code 2.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

```
[2/4] Fetching packages...
[1/5] Validating package.json...
[2/5] Resolving packages...
[3/5] Fetching packages...
error https://registry.yarnpkg.com/scrypt.js/-/scrypt.js-0.2.1.tgz: Extracting tar content of undefined failed, the file appears to be corrupt: "EEXIST: file already exists, mkdir '/root/.cache/yarn/v4/npm-scrypt-js-0.2.1-cc3f751933d6bac7a4bedf5301d7596e8146cdcd/node_modules/scrypt.js'"
info Visit https://yarnpkg.com/en/docs/cli/install for documentation about this command.
Same user, skipping permission fix
make: *** [hub-node-modules] Error 1
error Command failed with exit code 2.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

The errors appear intermittent and some combination of purging, clearing yarn cache, and simply retrying makes them go away usually. However, this is an annoying issue and needs to be addressed.

**Comment by ArjunBhuptani** [comment]
More yarn issues:
```
[Makefile] => Start building hub-node-modules
docker run --name=connext_buidler --tty --rm  --volume=/home/arjun/repositories/indra/modules/hub:/root connext_builder:dev 1000:1000 "yarn install --network-timeout 1000000"
Running command as user 1000:1000
yarn install v1.12.3
[1/4] Resolving packages...
[2/4] Fetching packages...
warning Pattern ["types-bn@https://github.com/machinomy/types-bn.git"] is trying to unpack in the same destination "/usr/local/share/.cache/yarn/v4/npm-types-bn-0.0.1-08ede69e138640d332b0dae0dc2e2fffee5f7160/node_modules/types-bn" as pattern ["types-bn@^0.0.1"]. This could result in non-deterministic behavior, skipping.
[1/5] Validating package.json...
[2/5] Resolving packages...
[3/5] Fetching packages...
error https://github.com/machinomy/types-bn.git: Extracting tar content of undefined failed, the file appears to be corrupt: "Invalid tar header. Maybe the tar is corrupted or it needs to be gunzipped?"
info Visit https://yarnpkg.com/en/docs/cli/install for documentation about this command.
Fixing permissions for 1000:1000
Makefile:132: recipe for target 'hub-node-modules' failed
make: *** [hub-node-modules] Error 1
error Command failed with exit code 2.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

**Comment by LayneHaber** [comment]
I've seen various yarn issues, they specifically pop up when the hub-node-modules and wallet-node-modules are building

```=============
[Makefile] => Start building hub-node-modules
docker run --name=connext_buidler --tty --rm --user 501:20 --volume=/Users/laynehaber/Developer/connext/repos/indra/modules/hub:/root connext_builder:dev 501:20 "yarn install --network-timeout 1000000"
Running command as user 501:20
yarn install v1.12.3
info No lockfile found.
[1/4] Resolving packages...
warning connext > web3 > web3-bzz > swarm-js > fs-promise@2.0.3: Use mz or fs-extra^3.0 with Promise Support
warning connext > web3 > web3-bzz > swarm-js > tar.gz@1.0.7: ⚠️  WARNING ⚠️ tar.gz module has been deprecated and your application is vulnerable. Please use tar module instead: https://npmjs.com/tar
warning machinomy > @types/commander@2.12.2: This is a stub types definition for commander (https://github.com/tj/commander.js). commander provides its own type definitions, so you don't need @types/commander installed!
warning sinon > nise > text-encoding@0.6.4: no longer maintained
warning sinon > @sinonjs/formatio > samsam@1.3.0: This package has been deprecated in favour of @sinonjs/samsam
[2/4] Fetching packages...
[1/5] Validating package.json...
[2/5] Resolving packages...
[3/5] Fetching packages...
[4/5] Linking dependencies...
[5/5] Building fresh packages...
$ rm -f node_modules/web3/index.d.ts
$ npm run build
npm WARN lifecycle The node binary used for scripts is /tmp/yarn--1546549473114-0.47812009183698145/node but npm is using /usr/local/bin/node itself. Use the `--scripts-prepend-node-path` option to in
clude the path for the node binary npm was executed with.

> connext-client@0.0.1 build /root/.cache/yarn/v4/.tmp/f3f5561d9073752ed3bd91d0ff442da6.42aac87eb270065814489f91a82d0130217ad300.prepare
> tsc -p tsconfig.json

info fsevents@1.2.4: The platform "linux" is incompatible with this module.
info "fsevents@1.2.4" is an optional dependency and failed compatibility check. Excluding it from installation.
[3/4] Linking dependencies...
error An unexpected error occurred: "ENOENT: no such file or directory, lstat '/root/.cache/yarn/v4/npm-commander-2.17.1-bd77ab7de6de94205ceacc72f1716d29f20a77bf/node_modules/commander/'".
info If you think this is a bug, please open a bug report with the information provided in "/root/yarn-error.log".
info Visit https://yarnpkg.com/en/docs/cli/install for documentation about this command.
Same user, skipping permission fix
make: *** [hub-node-modules] Error 1
error Command failed with exit code 2.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.

**Comment by LayneHaber** [comment]
Using `yarn start` and `yarn restart` after failures typically works to move past this

**Comment by bohendo** [comment]
For more info: https://github.com/yarnpkg/yarn/issues/2629

This issue should have been resolved while migrating to npm during PR #43 

Reopen if this problem resurfaces.
